plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -1988,6 +1990,35 @@ public void testMapMaxReadBytes()
         tester.testMaxReadBytes(getStandardMapObjectInspector(javaStringObjectInspector, javaLongObjectInspector), values, values, mapType(VARCHAR, BIGINT), maxReadBlockSize);
     }
 
+    @Test
+    public void testBackwardsCompatibleRepeatedStringField()
+            throws Exception
+    {
+        File parquetFile = new File(Resources.getResource("parquet_repeated_primitives/string/old-repeated-string.parquet").toURI());
+        List<List<String>> expectedValues = ImmutableList.of(Arrays.asList("hello", "world"), Arrays.asList("good", "bye"), Arrays.asList("one", "two", "three"));
+        tester.testReadPageOnly(parquetFile, ImmutableList.of("myString"), new ArrayType(VARCHAR), expectedValues);
+    }
+
+    @Test
+    public void testBackwardsCompatibleRepeatedIntegerField()
+            throws Exception
+    {
+        File parquetFile = new File(Resources.getResource("parquet_repeated_primitives/int/old-repeated-int.parquet").toURI());
+        List<List<Integer>> expectedValues = ImmutableList.of(Arrays.asList(1, 2, 3));
+        tester.testReadPageOnly(parquetFile, ImmutableList.of("repeatedInt"), new ArrayType(INTEGER), expectedValues);
+    }
+
+    @Test
+    public void testBackwardsCompatibleRepeatedPrimitiveFieldDefinedAsPrimitive()
+    {
+        assertThatThrownBy(() -> {
+            File parquetFile = new File(Resources.getResource("parquet_repeated_primitives/int/old-repeated-int.parquet").toURI());
+            List<List<Integer>> expectedValues = ImmutableList.of(Arrays.asList(1, 2, 3));
+            tester.testReadPageOnly(parquetFile, ImmutableList.of("repeatedInt"), INTEGER, expectedValues);
+        }).hasMessage("Unsupported Trino column type (integer) for Parquet column ([repeatedint] repeated int32 repeatedint)")
+                .isInstanceOf(TrinoException.class);
+    }
```

Comment:
These tests should go into trino-parquet module rather than trino-hive as we don't use hive/hadoop libs to write the data here. You can either add a new class or add these to `io.trino.parquet.reader.TestParquetReader`

Reply:
Sure, moved the tests from `AbstractTestParquetReader` and the helper method in `ParquetTester` in trino-hive to `TestParquetReader` in trino-parquet.